ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc

```diff
@@ -1113,19 +1113,21 @@ TFLiteCore::setInputTensorDim (const GstTensorsInfo *info)
 int
 TFLiteCore::reloadInterpreter (TFLiteInterpreter * new_interpreter)
 {
-  TFLiteInterpreter * interpreter_temp = interpreter;
-  if (!gst_tensors_info_is_equal (interpreter->getInputTensorsInfo (),
-          new_interpreter->getInputTensorsInfo ())
-      || !gst_tensors_info_is_equal (interpreter->getOutputTensorsInfo (),
-             new_interpreter->getOutputTensorsInfo ())) {
+  gboolean in_matched, out_matched;
+
+  interpreter->lock ();
+  in_matched = gst_tensors_info_is_equal (interpreter->getInputTensorsInfo (),
+      new_interpreter->getInputTensorsInfo ());
+  out_matched = gst_tensors_info_is_equal (interpreter->getOutputTensorsInfo (),
+      new_interpreter->getOutputTensorsInfo ());
+  interpreter->unlock ();
+
+  if (!in_matched || !out_matched) {
     ml_loge ("The model has unmatched tensors info\n");
     return -EINVAL;
   }
 
-  interpreter_temp->lock ();
   interpreter = new_interpreter;
```

Comment:
What if some other thread was using values of interpreter with interpreter->lock()?

Don't you need to unlock after LINE 1130?

Reply:
set new/old interpreter lock in this method.